code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -229,6 +234,22 @@ mob/living/simple_animal/hostile/hitby(atom/movable/AM as mob|obj,var/speed = TH
 	target_mob = null
 	return
 
+/mob/living/simple_animal/hostile/proc/Check_fire(target_mob)

```

Comment:
A bit of a minor quibble, but this is probably the weirdest naming style you can have. It's a mix of Pascal case and snake case. Please pick one or the other, not both.